src/implementations/vanilla/components/inpage-navigation/inpage-navigation.scss

```diff
@@ -0,0 +1,231 @@
+/**
+ * ECL Navigation inpage
+ * @define inpage-navigation
+ */
+
+@use "sass:map";
+@use '@ecl/theme-dev/maps/color';
+@use '@ecl/theme-dev/maps/icon';
+@use '@ecl/theme-dev/maps/spacing';
+@use '@ecl/theme-dev/maps/typography';
+@use '@ecl/vanilla-layout-grid/mixins/breakpoints';
+
+$_inpage-link-border: map.get(spacing.$spacing, '2xs');
+$_toggle-width: calc(
+  #{map.get(spacing.$spacing, 'm')} + #{map.get(icon.$icon, 's')} + #{map.get(
+      spacing.$spacing,
+      'm'
+    )}
+);
+
+.ecl-inpage-navigation {
+  display: none;
+  margin: 0;
+  position: relative;
+  z-index: 1;
+
+  /* stylelint-disable-next-line order/order */
+  @include breakpoints.up('l') {
+    display: block;
+    position: sticky;
+    top: 0;
+  }
+}
+
+.ecl-inpage-navigation__trigger {
+  background: linear-gradient(
+    to left,
+    map.get(color.$color, 'blue-120') $_toggle-width,
+    map.get(color.$color, 'primary') $_toggle-width
+  );
+  border: 0;
+  color: map.get(color.$color, 'white-100');
+  display: none;
+  padding: map.get(spacing.$spacing, 's') map.get(spacing.$spacing, 'm');
+  position: relative;
+  text-align: left;
+  width: 100%;
+
+  /* stylelint-disable-next-line order/order */
+  @include breakpoints.up('l') {
+    display: none;
+  }
+
+  &:focus {
+    outline: none;
+  }
+}
+
+.ecl-inpage-navigation--visible {
+  display: block;
+
+  .ecl-inpage-navigation__trigger {
+    display: block;
+    overflow-x: hidden;
+    padding-right: calc(#{$_toggle-width} + #{map.get(spacing.$spacing, 'm')});
+    text-overflow: ellipsis;
+    white-space: nowrap;
+
+    /* stylelint-disable-next-line order/order */
+    @include breakpoints.up('l') {
+      display: none;
+    }
+  }
+}
+
+.ecl-inpage-navigation__trigger-current {
+  font: map.get(typography.$font-prolonged, 'm');
+  font-weight: map.get(typography.$font-weight, 'bold');
+  margin-right: map.get(typography.$font, 'xl');
```

Comment:
The original value here is `margin-right: $ecl-font-size-xl;`.
The `typography.$font` carries font-weight instruction in addition to the size, so it overrides the rule for making the text bold.
On another note, although typography-related settings are migrated here as in original code, it's strange to see a font-related setting used for actual spacing instruction.

Reply:
Thks, nice catch ! After verification, as the label of mobile navigation is truncated in JS, there is no need to add an additional margin-right, the display is correct in mobile without this margin, so I removed the property 👍